cranelift/codegen/src/isa/x64/inst/emit.rs

```diff
@@ -3551,6 +3556,16 @@ pub(crate) fn emit(
                     sink.put1(0xc0 | (enc_dst & 7));
                     sink.put4(*offset as u32);
                 }
+            } else if distance == &RelocDistance::Near {
```

Comment:
Maybe put this before is_pic? All code in the same object file has a fixed relative location with each other.